autoscale/src/main/java/com/sequenceiq/periscope/controller/DistroXAutoScaleScalingActivityV1Controller.java

```diff
@@ -0,0 +1,119 @@
+package com.sequenceiq.periscope.controller;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceCrn;
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceName;
+import com.sequenceiq.authorization.annotation.DisableCheckPermissions;
+import com.sequenceiq.authorization.annotation.ResourceCrn;
+import com.sequenceiq.authorization.annotation.ResourceName;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.periscope.api.endpoint.v1.DistroXAutoScaleScalingActivityV1Endpoint;
+import com.sequenceiq.periscope.api.model.DistroXAutoscaleScalingActivityResponse;
+import com.sequenceiq.periscope.converter.DistroXAutoscaleScalingActivityResponseConverter;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.model.NameOrCrn;
+import com.sequenceiq.periscope.service.ScalingActivityService;
+
+@Controller
+public class DistroXAutoScaleScalingActivityV1Controller implements DistroXAutoScaleScalingActivityV1Endpoint {
+
+    @Inject
+    private DistroXAutoscaleScalingActivityResponseConverter distroXAutoscaleScalingActivityResponseConverter;
+
+    @Inject
+    private ScalingActivityService scalingActivityService;
+
+    @Inject
+    private AutoScaleClusterCommonService asClusterCommonService;
+
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.SCALE_DATAHUB)
+    public List<DistroXAutoscaleScalingActivityResponse> getAllScalingActivityByClusterName(@ResourceName String clusterName,
+            Integer pageNumber, Integer pageSize) {
+        Cluster cluster = asClusterCommonService.getClusterByCrnOrName(NameOrCrn.ofName(clusterName));
+        Pageable pageable = PageRequest.of(pageNumber, pageSize);
+        return distroXAutoscaleScalingActivityResponseConverter.convertAllToJson(scalingActivityService.findAllForCluster(cluster.getId(), pageable));
+    }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.SCALE_DATAHUB)
+    public List<DistroXAutoscaleScalingActivityResponse> getAllScalingActivityByClusterCrn(@ResourceCrn String clusterCrn,
+            Integer pageNumber, Integer pageSize) {
+        Cluster cluster = asClusterCommonService.getClusterByCrnOrName(NameOrCrn.ofCrn(clusterCrn));
+        Pageable pageable = PageRequest.of(pageNumber, pageSize);
+        return distroXAutoscaleScalingActivityResponseConverter.convertAllToJson(scalingActivityService.findAllForCluster(cluster.getId(), pageable));
+    }
+
+    @Override
+    @DisableCheckPermissions
```

Comment:
This has to be changed to the applicable authz annotation - `@CheckPermissionByResourceCrn`. Debugging via `PermissionCheckService` should provide insights on any authz issues

Reply:
Done